src/sliceDataStorage.cpp

```diff
@@ -268,23 +203,11 @@ bool SliceMeshStorage::getExtruderIsUsed(const size_t extruder_nr, const LayerIn
             }
         }
     }
-    if (settings.get<bool>("fill_outline_gaps")
-        && settings.get<size_t>("wall_line_count") > 0
-        && settings.get<ExtruderTrain&>("wall_0_extruder_nr").extruder_nr == extruder_nr)
-    {
-        for (const SliceLayerPart& part : layer.parts)
-        {
-            if (part.outline_gaps.size() > 0)
-            {
-                return true;
-            }
-        }
-    }
     if ((settings.get<size_t>("wall_line_count") > 1 || settings.get<bool>("alternate_extra_perimeter")) && settings.get<ExtruderTrain&>("wall_x_extruder_nr").extruder_nr == extruder_nr)
     {
         for (const SliceLayerPart& part : layer.parts)
         {
-            if (part.insets.size() > 1 && part.insets[1].size() > 0)
+            if (!part.wall_toolpaths.empty() && part.hasWallAtInsetIndex(1))
```

Comment:
`hasWallAtInsetIndex(1)` already includes checking if `wall_toolpaths` is empty.
```suggestion
            if(part.hasWallAtInsetIndex(1))
```